protocol/x/subaccounts/keeper/subaccount.go

```diff
@@ -515,6 +522,39 @@ func (k Keeper) internalCanUpdateSubaccounts(
 	success = true
 	successPerUpdate = make([]types.UpdateResult, len(settledUpdates))
 
+	// Block all withdrawals and transfers if there was a negative TNC subaccount seen within the
+	// last `WITHDRAWAL_AND_TRANSFERS_BLOCKED_AFTER_NEGATIVE_TNC_SUBACCOUNT_SEEN_BLOCKS`.
+	if updateType == types.Withdrawal || updateType == types.Transfer {
```

Comment:
nit: can we move this to a helper function?

Reply:
LMK if you hard disagree here, but this logic is pretty simple for now so I prefer to leave it in the main function for readability

I agree once we add more reasons for blocking withdrawals / trades / etc then all of that should be abstracted into a helper function

LMK if I'm missing your point though